common/frame_manager.go

```diff
@@ -117,7 +117,14 @@ func (m *FrameManager) frameAbortedNavigation(frameID cdp.FrameID, errorText, do
 	defer m.framesMu.Unlock()
 
 	frame := m.frames[frameID]
-	if frame == nil || frame.pendingDocument == nil {
+	if frame == nil {
+		return
+	}
+
+	frame.pendingDocumentMu.Lock()
+	defer frame.pendingDocumentMu.Unlock()
```

Comment:
We're emitting an event that contains the locked document near the end of the `frameAbortedNavigation` method:

https://github.com/grafana/xk6-browser/blob/9047580e50c5bba56ef3c6d0ce268ea7dfd8f6ea/common/frame_manager.go#L138

And the following code accesses the locked document. May it cause a deadlock? 🤔 

https://github.com/grafana/xk6-browser/blob/9047580e50c5bba56ef3c6d0ce268ea7dfd8f6ea/common/frame_manager.go#L568-L573

If there is a deadlock possibility, wdyt about unlocking the mutex before emitting the event or cloning the document? 🤔

Reply:
Yep, good suggestions.

Resolved in: 544c4f17dca3e95976d30f5adb6fe0befb86db80 & f83aa3bad979c5ead203b19c50bba39ecdf76fbb

Reply:
Yeah, that's interesting actually. We're not using the mutex when we read the document from the event, which i think should be fine, since all the consumers of `EventFrameNavigation` only read and don't try to mutate that document.

Nevertheless, it's good practice to give up the lock as early as possible, and I'm happy to do that for this case.

Resolved in: 9d8ce43b7a9d876e5483ba98e830c945af38c23e